### PR TITLE
[verifier] Minor hardening against arithmetic overflow

### PR DESCRIPTION
**Pull request by wrwg**
## Motivation

Hardening

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

Yes

## Test Plan

NA

### CODE REVIEW

**Review by meng-xu-cs — APPROVED** [review]

File: language/move-vm/transactional-tests/tests/module_publishing/republish_module_skip_compatible_linking_hack_struct.exp [hunk 1]
```diff
@@ -1 +1,28 @@
 processed 5 tasks
+
```
Comment (tnowacki) [inline language/move-vm/transactional-tests/tests/module_publishing/republish_module_skip_compatible_linking_hack_struct.exp:2]: Is this test change related?

Comment (wrwg) [reply language/move-vm/transactional-tests/tests/module_publishing/republish_module_skip_compatible_linking_hack_struct.exp:2]: Wow thats bad. Just seeing now. Lets followup on this.
